### PR TITLE
fail build if plugin cant load next.config.js

### PR DESCRIPTION
**Pull request by lindsaylevine**
Fixes https://github.com/netlify/netlify-plugin-nextjs/issues/78
Fixes https://github.com/netlify/netlify-plugin-nextjs/issues/79

not 100% sure about trying to pass in the exact error from the catch into `failBuild`, open to thoughts!

### CODE REVIEW

File: helpers/hasCorrectNextConfig.js [hunk 1]
```diff
@@ -12,7 +12,12 @@ const hasCorrectNextConfig = (nextConfigPath) => {
 
   // If the next config exists, log warning if target isnt in acceptableTargets
   const acceptableTargets = ['serverless', 'experimental-serverless-trace']
-  const nextConfig = loadConfig(PHASE_PRODUCTION_BUILD, path.resolve('.'))
+  let nextConfig
+  try {
+    nextConfig = loadConfig(PHASE_PRODUCTION_BUILD, path.resolve('.'))
+  } catch (e) {
+    return failBuild('Error loading your next.config.js.')
+  }
```
Comment (erezrokah) [inline helpers/hasCorrectNextConfig.js:20]:
```suggestion
  } catch (error) {
    return failBuild('Error loading your next.config.js.', { error })
  }
```

File: helpers/doesNotNeedPlugin.js [hunk 2]
```diff
@@ -5,15 +5,15 @@ const isStaticExportProject = require('./isStaticExportProject')
 const doesSiteUseNextOnNetlify = require('./doesSiteUseNextOnNetlify')
 const hasCorrectNextConfig = require('./hasCorrectNextConfig')
 
-const doesNotNeedPlugin = async ({ netlifyConfig, packageJson }) => {
+const doesNotNeedPlugin = async ({ netlifyConfig, packageJson, utils }) => {
```
Comment (erezrokah) [inline helpers/doesNotNeedPlugin.js:8]: [dust] should we pass only `failBuild`

**Review by erezrokah — CHANGES_REQUESTED** [review]
👍 For more error handling.

I think we should pass the original error as suggested [here](https://docs.netlify.com/configure-builds/build-plugins/create-plugins/#error-reporting) to make it easier for users to self serve

**Review by ehmicky — COMMENTED** [review]
Thanks @lindsaylevine!

File: helpers/doesNotNeedPlugin.js (hunk 2 above)
Comment (lindsaylevine) [reply helpers/doesNotNeedPlugin.js:8]: i was trying to keep it on the same level as netlifyConfig and packageJson, as in they're all things that get passed to the different event handlers (i.e onBuild) so i think it's ok? will keep in mind tho going forward

**Pull request merged**